tensorflow_model_optimization/python/core/quantization/keras/prune_preserve/prune_preserve_quantize_registry.py

```diff
@@ -0,0 +1,332 @@
+# limitations under the License.
+# ==============================================================================
+"""Registry responsible for built-in keras classes."""
+
+import tensorflow as tf
+
+from tensorflow.python.keras.engine.base_layer import TensorFlowOpLayer
+
+from tensorflow_model_optimization.python.core.quantization.keras import quant_ops
+from tensorflow_model_optimization.python.core.quantization.keras import quantizers
+from tensorflow_model_optimization.python.core.quantization.keras.default_8bit import (
+    default_8bit_quantizers)
+
+layers = tf.keras.layers
+
+
+class _PrunePreserveInfo(object):
+  """PrunePreserveInfo."""
+  def __init__(self, weight_attrs, quantize_config_attrs):
+    """PrunePreserveInfo.
+
+    Args:
+      weight_attrs: list of sparsity preservable weight attributes of layer.
+      quantize_config_attrs: list of quantization configuration class name.
+    """
+    self.weight_attrs = weight_attrs
+    self.quantize_config_attrs = quantize_config_attrs
+
+
+def _noPrunePreserve():
+  return _PrunePreserveInfo([], [])
+
+
+class PrunePreserveQuantizeRegistry(object):
+  """PrunePreserveQuantizeRegistry responsible for built-in keras layers."""
+
+  # The keys represent built-in keras layers; the first values represent the
+  # the variables within the layers which hold the kernel weights, second
+  # values represent the class name of quantization configuration for layers.
+  # This decide the weights of layers with quantization configurations are
+  # sparsity preservable.
+  _LAYERS_CONFIG_MAP = {
+      layers.Conv2D:
+      _PrunePreserveInfo(['kernel'], ['Default8BitConvQuantizeConfig']),
+      layers.Dense:
+      _PrunePreserveInfo(['kernel'], ['Default8BitQuantizeConfig']),
+
+      # layers that supported with prune, but not yet with qat
+      # layers.Conv1D:
+      # _PrunePreserveInfo(['kernel'], []),
+      # layers.Conv2DTranspose:
+      # _PrunePreserveInfo(['kernel'], []),
+      # layers.Conv3D:
+      # _PrunePreserveInfo(['kernel'], []),
+      # layers.Conv3DTranspose:
+      # _PrunePreserveInfo(['kernel'], []),
+      # layers.LocallyConnected1D:
+      # _PrunePreserveInfo(['kernel'], ['Default8BitQuantizeConfig']),
+      # layers.LocallyConnected2D:
+      # _PrunePreserveInfo(['kernel'], ['Default8BitQuantizeConfig']),
+
+      # DepthwiseCon2D is supported with 8bit qat, but not with prune
+      # layers.DepthwiseConv2D:
+      # _PrunePreserveInfo(['depthwise_kernel'], ['Default8BitConvQuantizeConfig']),
+
+      # SeparableConv need verify from 8bit qat
+      # layers.SeparableConv1D:
+      # _PrunePreserveInfo(['pointwise_kernel'], ['Default8BitConvQuantizeConfig']),
+      # layers.SeparableConv2D:
+      # _PrunePreserveInfo(['pointwise_kernel'], ['Default8BitConvQuantizeConfig']),
+
+      # Embedding need verify from 8bit qat
+      # layers.Embedding: _PrunePreserveInfo(['embeddings'], []),
+
+      layers.ELU: _noPrunePreserve(),
```

Comment:
You can use the approach used by Elena under https://github.com/tensorflow/model-optimization/pull/539 to cut almost all of the _noPrunePreserve lines.

Reply:
thanks!
done, please have a review.